oioswift/common/middleware/container_hierarchy.py

```diff
@@ -43,6 +43,55 @@
 
 
 class RedisDb(object):
+    lua_script_hkeys = """
+        local get_index = function(key, start, delimiter)
+            -- Get index of the first delimiter of the key
+            local index = string.find(key, delimiter, start)
+            if index  == nil then
+                return #key
+            end
+            return index
```

Comment:
> `string.find (s, pattern [, init [, plain]])` Looks for the first match of pattern in the string `s`. If it finds a match, then find returns the indices of `s` where this occurrence starts and ends; otherwise, it returns `nil`.

We should only keep the first index. But that does not seem to be a problem for `string.sub`.